node/src/perf_config.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+
+use graph::{anyhow, prelude::regex::Regex};
+use graph_core::{SubgraphPerfConfig, SubgraphPerfRules};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SubgraphConfigSection {
+    rules: HashMap<String, SubgraphConfigRule>,
+}
+
+impl TryInto<SubgraphPerfRules> for SubgraphConfigSection {
+    type Error = anyhow::Error;
+
+    fn try_into(self) -> Result<SubgraphPerfRules, Self::Error> {
+        self.rules
+            .into_values()
+            .map(|x| {
+                let key = match x.match_rule {
+                    MatchSelection::SubgraphName { name } => Regex::new(&name),
+                };
+
+                key.map(|k| (k, x.actions.into()))
+                    .map_err(anyhow::Error::from)
+            })
+            .collect::<anyhow::Result<Vec<(Regex, SubgraphPerfConfig)>>>()
+            .map(|rules| SubgraphPerfRules { rules })
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SubgraphConfigRule {
+    #[serde(alias = "match")]
+    match_rule: MatchSelection,
+    actions: SubgraphConfigDetails,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct SubgraphConfigDetails {
+    history_blocks: i32,
+}
+
+impl Into<SubgraphPerfConfig> for SubgraphConfigDetails {
+    fn into(self) -> SubgraphPerfConfig {
+        let Self { history_blocks } = self;
+
+        SubgraphPerfConfig { history_blocks }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum MatchSelection {
+    SubgraphName { name: String },
+}
+
+#[cfg(test)]
+mod test {
+    use crate::perf_config::MatchSelection;
+
+    #[test]
+    fn parses_correctly() {
+        let content = r#"
+        [rules]
+
+        [rules.name] 
+        match = { type = "subgraph_name", name = ".*" } 
```

Comment:
I don't understand why we need both `type` and `name` here. That seems a bit cumbersome and just having `match { name = "foo.*" }` should be enough.

Reply:
I was thinking later we could add an owner match instead of just name